server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1496,7 +1523,7 @@ public void flush(FlushRequest request) {
      * {@link org.opensearch.index.translog.TranslogDeletionPolicy} for details
      */
     public void trimTranslog() {
-        if (isRemoteTranslogEnabled()) {
+        if (isRemoteTranslogEnabled() || isMigratingToRemote()) {
```

Comment:
lets replace this `if (isRemoteTranslogEnabled() || isMigratingToRemote()) {`  with `indexSettings.isRemoteNode()`?